T5661: Add show ssh dynamic-protection and show log ssh dynamic-protection

JeffWDH opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary

Adds functionality to show IPs blocked by SSH dynamic-protection as well as the SSHGuard logs.
Manual backport of #2369

vyos-documentation PR 1127 has been submitted to update the documentation.

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://vyos.dev/T5661

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->

## Proposed changes
<!--- Describe your changes in detail -->
Add:
show ssh dynamic-protection
show log ssh dynamic-protection

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->

## Smoketest result
<!-- Provide the output of the smoketest
```
$ /usr/libexec/vyos/tests/smoke/cli/test_xxx_feature.py
test_01_simple_options (__main__.TestFeature.test_01_simple_options) ... ok
```
-->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

Commit:
T5661: Add show ssh dynamic-protection and show log ssh dynamic-protection

c-po commented on src/op_mode/ssh.py:
```diff
@@ -60,3 +61,40 @@ def show_fingerprints(raw: bool, ascii: bool):
+    attackers = []
+    try:
+        # IPv4
+        attackers = attackers + json.loads(cmd("sudo nft -j list set ip sshguard attackers"))["nftables"][1]["set"]["elem"]
```
We try to not have `sudo` in the op mode scripts as long as this is not ultimately required. Please rather add the `sudo` call to the OP mode XML template.

JeffWDH replied:
I see calls to sudo in many other of the Python op_mode scripts... This script does more than just show dynamic-protection stuff, such as showing SSH public key fingerprints, which does not require root. I think limiting the use of root to where it is only needed (showing the tables) makes the most security sense. 

Also, this was accepted as is to 1.5, do we really need to change it? If so, it should be updated there as well.

c-po replied:
We have a lot of code accepted in 1.4 and 1.5 that we later iterate on. No big deal.

I will append a commit with the changes to this PR here

c-po replied:
The remaining calls to `sudo` are there b/c those scripts have yet most likely not been converted to the new style op-mode scripts.

c-po commented on op-mode-definitions/show-log.xml.in:
```diff
@@ -398,12 +398,20 @@
+                <properties>
+                  <help>Show dynamic-protection log</help>
+                </properties>
+                <command>journalctl --no-hostname -p info -t sshguard -o short</command>
```
Any particular reason to diverge from the usual path by specifying the systemd unit to read from?

Should be: `journalctl --no-hostname --boot --unit sshguard.service` instead

JeffWDH replied:
This was accepted like this on the 1.5 commit, does it need to change? The output is functionally the same. If so, like the sudo change, 1.5 would need to be updated as well.

c-po replied:
We have a lot of code accepted in 1.4 and 1.5 that we later iterate on. No big deal.

I will append a commit with the changes to this PR here

c-po commented on op-mode-definitions/show-ssh.xml.in:
```diff
@@ -7,6 +7,12 @@
+            <properties>
+              <help>Show SSH server dynamic-protection blocked attackers</help>
+            </properties>
+            <command>${vyos_op_scripts_dir}/ssh.py show_dynamic_protection</command>
```
Add `sudo` here and see other comment about sudo below.

c-po committed:
op-mode: T5661: use common journalctl syntax for sshguard

This makes the code more easy to maintain in the future if everyone uses the
same structure when calling journalctl.

(cherry picked from commit e1b4e972b40941acec76c97e714767214cefe426)

c-po committed:
op-mode: T5661: remove call to sudo in ssh.py and move it to XML definition

Try to have as few calls to sudo in the op-mode scripts as possible. The XML
definitions can deal with it.

(cherry picked from commit 428dee29d36cc3629990ec41afef887821886834)

c-po committed:
op-mode: T5661: add "monitor ssh dynamic-protection" command to follow the logfile

(cherry picked from commit 78e00bf4099bfac2164ef2075acce8169c40c9c3)
